### DISCUSSION TITLE
SetTimeout doesn't work in next api

### DISCUSSION THREAD

**Discussion by rodriguesvinicius** [vercel/next.js#38746] [Help]
I have a problem, that when I call my endpoint it has a setTimeout that needs to be executed in parallel, but returning 200 ok in the api, what happens is that the endpoint is not processing the code inside the setTimeout it ends the endpoint and completely ignores it, which I found strange and when I call the endpoint again it kind of saved the execution of the code from the first call to the endpoint, I need it to execute the code in the first call not influencing the second call

![image](https://user-images.githubusercontent.com/53584452/179444034-bcce13cd-c9c6-4fb9-97cf-86fddcf94874.png)

**Comment by icyJoseph** [comment]
I don't understand, you want to answer with `ok`, and also run a timeout, but the moment you call `.send` you end the communication with the client.

That's also a huge timeout, I think the Vercel API might kill that lambda before the timeout is executed, https://vercel.com/docs/concepts/limits/overview#serverless-function-execution-timeout

Perhaps you want to do.

```js
setTimeout(() => {
   MercadoPago.payment.search().then().catch().finally(()=> res.send('Ok'));
}, 20_000)
```

I omitted a bunch of other things, but the idea is to respond at the end of the async code.

This feature feels strange though, so I'll not make any assumptions, or comments on other things I find strange to not leak my bias into it.
